src/lm/searchTools.ts

```diff
@@ -0,0 +1,439 @@
+ *--------------------------------------------------------------------------------------------*/
+'use strict';
+
+import * as vscode from 'vscode';
+import Logger from '../common/logger';
+import { FolderRepositoryManager } from '../github/folderRepositoryManager';
+import { ILabel, Issue } from '../github/interface';
+import { RepositoriesManager } from '../github/repositoriesManager';
+import { ChatParticipantState } from './participants';
+import { concatAsyncIterable, ToolBase } from './tools/toolsUtils';
+
+interface ConvertToQuerySyntaxParameters {
+	naturalLanguageString: string;
+	repo?: {
+		owner?: string;
+		name?: string;
+	};
+}
+
+interface ConvertToQuerySyntaxResult {
+	query: string;
+	repo?: {
+		owner?: string;
+		name?: string;
+	};
+}
+
+enum ValidatableProperty {
+	is = 'is',
+	type = 'type',
+	state = 'state',
+	in = 'in',
+	linked = 'linked',
+	status = 'status',
+	draft = 'draft',
+	review = 'review',
+	no = 'no',
+}
+
+const MATCH_UNQUOTED_SPACES = /(?!\B"[^"]*)\s+(?![^"]*"\B)/;
+
+export class ConvertToSearchSyntaxTool extends ToolBase<ConvertToQuerySyntaxParameters> {
+	static ID = 'ConvertToSearchSyntaxTool';
+	constructor(private readonly repositoriesManager: RepositoriesManager, chatParticipantState: ChatParticipantState) {
+		super(chatParticipantState);
+	}
+
+	private async fullQueryAssistantPrompt(folderRepoManager: FolderRepositoryManager): Promise<string> {
+		const remote = folderRepoManager.activePullRequest?.remote ?? folderRepoManager.activeIssue?.remote ?? (await folderRepoManager.getPullRequestDefaultRepo()).remote;
+
+		return `Instructions:
+You are an expert on GitHub issue search syntax. GitHub issues are always software engineering related. You can help the user convert a natural language query to a query that can be used to search GitHub issues. Here are some rules to follow:
+- Always try to include "repo:" or "org:" in your response.
+- "repo" is often formated as "owner/name". If needed, the current repo is ${remote.owner}/${remote.repositoryName}.
+- Ignore display information.
+- Respond with only the query.
+- Always include a "sort:" parameter.
+- Always include a property with the @me value if the query includes "me" or "my".
+- Here are some examples of valid queries:
+	- repo:microsoft/vscode is:issue state:open sort:updated-asc
+	- mentions:@me org:microsoft is:issue state:open sort:updated
+	- assignee:@me milestone:"October 2024" is:open is:issue sort:reactions
+	- comments:>5 org:contoso is:issue state:closed mentions:@me label:bug
+	- interactions:>5 repo:contoso/cli is:issue state:open
+- Go through each word of the natural language query and try to match it to a syntax component.
+- As a reminder, here are the components of the query syntax:
+	Filters:
+| Property 	| Possible Values | Value Description |
+|-----------|-----------------|-------------------|
+| is 		| issue, pr, draft, public, private, locked, unlocked |  |
+| assignee 	|  | A GitHub user name or @me |
```

Comment:
I would expect something more like json or a TS type would work better here?

Reply:
Changed to json. No tests, so I don't actually know if it's better.